### PR TITLE
Add os.allocated_processors stats

### LINKED ISSUE #13917: processors setting is not reflected in nodes info API ("os.available_processors")
https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-threadpool.html#processors

The end of that page seems to imply the `processors` value passed in settings will be displayed in the nodes info API with the `os` flag. However, it seems to always display information retrieved from the OS.

I know the setting is going through due to the change in thread pool values.

Should the "available_processors" value display what is passed in settings, or is it assumed to be actual OS information?

(on 1.7.1)

### PR DESCRIPTION
**Pull request by xuzha**
Current processors setting is not reflected in nodes info API
("os.available_processors"). Add os.allocated_processors to shows
actual number of processors that we are using.

closes #13917

### CODE REVIEW

File: docs/reference/cluster/nodes-info.asciidoc [hunk 1]
```diff
@@ -54,6 +54,9 @@ the operating system:
 `os.available_processors`::
 	Number of processors available to the Java virtual machine
 
+`os.allocated_processors`::
+    Number of processors that ES is using. Imply the `processors` value passed in settings, at most 32
```
Comment (bleskes) [inline docs/reference/cluster/nodes-info.asciidoc]: Can we reword this to `The number of processors actually used to calculate thread pool size. This number can be set with the `processors` setting of a node and defaults to the number of processors reported by the OS. In both cases this number will never be larger than 32.

File: core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java [hunk 2]
```diff
@@ -337,6 +345,7 @@ public void readFrom(StreamInput in) throws IOException {
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             out.writeVInt(availableProcessors);
+            out.writeVInt(allocatedProcessors);
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: watch out with back porting..

**Comment by bleskes** [comment]
Left some minor comments. LGTM. I think this can go all the way down to 2.1, do you agree?

File: core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java (hunk 2 above)
Comment (xuzha) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java]: @bleskes thanks a lot. 

Hmm, I'm back porting this change to 2.1 and 2.x, but I don't see any conflict here. So could I ask what should I watch out here.

**Comment by xuzha** [comment]
Merged in master. 

Will back port this to 2.1 and 2.x tonight.

**Comment by xuzha** [comment]
Has been backported to 
2.1 : https://github.com/elastic/elasticsearch/commit/ba95190fdb1a633799aef98c6513ba3d4876bcaa
2.x : https://github.com/elastic/elasticsearch/commit/35e41d3d2ec055f6b17c867553cc38d1c8f9b8c8
